lib/reporting/reports/customers/base.rb

```diff
@@ -75,6 +74,10 @@ def filter_to_order_cycle(orders)
           end
         end
 
+        def last_completed_order(orders)
+          orders.max_by(&:completed_at)
+        end
+
         def last_completed_order_date(orders)
           orders.max_by(&:completed_at)&.completed_at&.to_date
```

Comment:
This should be using `last_completed_order` defined above

Reply:
Absolutely, that's why I've created this one ; sorry.